### ISSUE TITLE
[BUG] divide plugin  webclient throw exception: The connection observed an error,io.netty.handler.timeout.ReadTimeoutException: null

### ISSUE THREAD

**Issue by dongxl123** [apache/shenyu#2303]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

2021-11-04 10:58:29 [XNIO-1 I/O-3] INFO  org.apache.shenyu.plugin.httpclient.WebClientPlugin - The request urlPath is /api/v1/query?query=rate%28recv_packet_total%7Bsource%21%3D%22cffex%22%2Ctarget%3D~%22cffex.%2A%22%7D%5B5m%5D

%29&time=2021-11-04T02%3A58%3A28.564680692Z, retryTimes is 3
2021-11-04 10:58:32 [reactor-http-epoll-3] WARN  reactor.netty.http.client.HttpClientConnect - [id: 0xa0255f85, L:/10.201.220.93:34256 - R:product.thanos.10jqka.com.cn/172.20.208.115:80] The connection observed an error
io.netty.handler.timeout.ReadTimeoutException: null
2021-11-04 10:58:32 [reactor-http-epoll-3] ERROR org.apache.shenyu.plugin.httpclient.WebClientPlugin - null
io.netty.handler.timeout.ReadTimeoutException: null
2021-11-04 10:58:32 [reactor-http-epoll-3] ERROR org.apache.shenyu.web.handler.GlobalErrorHandler - [e9ba400] Resolved [ReadTimeoutException: null] for HTTP GET /prometheus/api/v1/query
2021-11-04 10:58:32 [reactor-http-epoll-3] ERROR org.springframework.boot.autoconfigure.web.reactive.error.AbstractErrorWebExceptionHandler - [e9ba400]  500 Server Errorfor HTTP GET "/prometheus/api/v1/query?query=rate

%28recv_packet_total%7Bsource%21%3D%22cffex%22%2Ctarget%3D~%22cffex.%2A%22%7D%5B5m%5D%29&time=2021-11-04T02%3A58%3A28.564680692Z"
io.netty.handler.timeout.ReadTimeoutException: null
2021-11-04 10:58:32 [reactor-http-epoll-3] ERROR reactor.core.publisher.Operators - Operator called default onErrorDropped
java.lang.NullPointerException: null
2021-11-04 10:58:32 [reactor-http-epoll-3] WARN  io.netty.channel.AbstractChannelHandlerContext - An exception 'reactor.core.Exceptions$BubblingException: java.lang.NullPointerException' [enable DEBUG level for full stacktrace] was 

thrown by a user handler's exceptionCaught() method while handling the following exception:
io.netty.handler.timeout.ReadTimeoutException: null
10:58:39,277 |-INFO in ReconfigureOnChangeTask(born:1634129949276) - Empty watch file list. Disabling

### Environment

```markdown
ShenYu version(s):2.4.0
```

**Comment by MarsOu1995** [comment]
I have the same problem.

**Comment by yu199195** [comment]
config this， or use 2.4.1

```
shenyu:
#  httpclient:
#    strategy: webClient
#    connectTimeout: 45000
#    readTimeout: 3000
#    writeTimeout: 3000
#    wiretap: false
#    pool:
#      type: ELASTIC
#      name: proxy
#      maxConnections: 16
#      acquireTimeout: 45000
#    proxy:
#      host:
#      port:
#      username:
#      password:
#      nonProxyHostsPattern:
#    ssl:
#      useInsecureTrustManager: false
#      trustedX509Certificates:
#      handshakeTimeout:
#      closeNotifyFlushTimeout:
#      closeNotifyReadTimeout:
#      defaultConfigurationType:
```

**Comment by yu199195** [comment]
see this : https://github.com/apache/incubator-shenyu/issues/1730

**Comment by shenzulun** [comment]
> config this， or use 2.4.1
> 
> ```
> shenyu:
> #  httpclient:
> #    strategy: webClient
> #    connectTimeout: 45000
> #    readTimeout: 3000
> #    writeTimeout: 3000
> #    wiretap: false
> #    pool:
> #      type: ELASTIC
> #      name: proxy
> #      maxConnections: 16
> #      acquireTimeout: 45000
> #    proxy:
> #      host:
> #      port:
> #      username:
> #      password:
> #      nonProxyHostsPattern:
> #    ssl:
> #      useInsecureTrustManager: false
> #      trustedX509Certificates:
> #      handshakeTimeout:
> #      closeNotifyFlushTimeout:
> #      closeNotifyReadTimeout:
> #      defaultConfigurationType:
> ```

the latest version(2.4.2-SNAPSHOT) also has this same bug，I'll try with this configuration...

**Comment by qicz** [comment]
@shenzulun Have you solved it?

**Comment by haitun521** [comment]
2.4.3 also has this same bug

**Comment by Rubenmp** [comment]
Just in case it helps.

**I had the same error in a personal project and I was able to solve it.**
I was using _webClient_ ('org.springframework.boot:spring-boot-starter-webflux:2.7.1') for async calls and getting this annoying errors polluting stderr:
```
14:39:38.128 [reactor-http-nio-4] ERROR r.n.http.client.HttpClientConnect - [id: 0x05f49b52, L::60850 - R:] The connection observed an error
io.netty.handler.timeout.ReadTimeoutException: null
```
This code was throwing the errors:
```
return webClient.get().uri(url).retrieve().bodyToFlux(String.class)
        .timeout(timeout); // <---- The problem is here
```
but I solved it using a fallback:
```
return webClient.get().uri(url).retrieve().bodyToFlux(String.class)
        .timeout(timeout, e -> {
            // Log timeout exception here to prevent pollution of stderr
        });
```

**Why?**
Look at the different timeout methods in _reactor-core_:
```
public final Flux<T> timeout(Duration timeout);
public final Flux<T> timeout(Duration timeout, @Nullable Publisher<? extends T> fallback);
```
For some reason, if the fallback is not provided a default fallback is created polluting the output (I strongly disagree with that implementation choice). If we specify the fallback then there are no more errors like those 'io.netty.handler.timeout.ReadTimeoutException: null'.

I have searched in this project and there are several timeouts without fallbacks:
![timeoouts-in-project](https://user-images.githubusercontent.com/11651356/178029234-d189dd46-86f4-4da8-a748-6d9bbbf1206b.png)
I think that this issue can be solved by adding fallbacks. I can add empty fallbacks to solve the issue but maybe a maintainer with more experience in this project prefer another solution?
